extensions/amp-list/0.1/amp-list.js

```diff
@@ -710,8 +710,16 @@ export class AmpList extends AMP.BaseElement {
       itemsExpr,
       this.element
     );
-    if (this.element.hasAttribute('single-item') && !isArray(items)) {
-      items = [items];
+    if (this.element.hasAttribute('single-item')) {
+      if (!isArray(items)) {
+        items = [items];
+      } else {
+        user().warn(
+          TAG,
+          'items must be a non-array Object if "single-item" is enabled.',
```

Comment:
Made a suggestion to sound more like other warnings emitted in this class:
```suggestion
          'Expected response to contain a non-array Object due to "single-item" attribute.',
```